### PR TITLE
Fixed issue with openNI2 while handing both 32F and 8U depth formats

### PR DESCRIPTION
**Pull request by nlyubova**

### CODE REVIEW

File: src/linemod_detect.cpp [hunk 1]
```diff
@@ -277,10 +277,16 @@ struct Detector: public object_recognition_core::db::bases::ModelReaderBase {
       }
       if (*use_depth_)
       {
+        if (depth_->depth() == 5)
```
Comment (vrabaud) [inline src/linemod_detect.cpp]: that is CV_32F right ? If so, please use CV_32F

File: src/linemod_detect.cpp [hunk 2]
```diff
@@ -277,10 +277,16 @@ struct Detector: public object_recognition_core::db::bases::ModelReaderBase {
...
+        {
+          *depth_ *= 1000.0f;
+          depth_->convertTo(*depth_, CV_16UC1);
```
Comment (vrabaud) [inline src/linemod_detect.cpp]: convertTo can have a scale argument: please use to have everything in only one operation

Comment (nlyubova) [inline src/linemod_detect.cpp]: done!

File: src/linemod_detect.cpp (hunk 1 above)
Comment (nlyubova) [inline src/linemod_detect.cpp]: done!

File: src/linemod_detect.cpp [hunk 3]
```diff
@@ -277,10 +277,13 @@ struct Detector: public object_recognition_core::db::bases::ModelReaderBase {
       }
       if (*use_depth_)
       {
+        if (depth_->depth() == CV_32F)
+          depth_->convertTo(*depth_, CV_16UC1, 1000.0);
```
Comment (vrabaud) [inline src/linemod_detect.cpp]: actually, there is a problem with that: you are overwriting depth_ which can be shared with different cells. Please simple create a cv::Mat depth. Code should be:

    cv::Mat depth = *depth_;
    if (depth_->depth() == CV_32F)
      depth.convertTo(depth, CV_16UC1, 1000.0);

And make sure you are using that matrix all over below. Sorry to have missed that

**Comment by vrabaud** [comment]
fixing myself

**Comment by nlyubova** [comment]
Thank you for fixing!

**Pull request merged**